api/client/credentials.go

```diff
@@ -120,21 +137,36 @@ func (c *IdentityCreds) Dialer() (ContextDialer, error) {
 	return nil, trace.NotImplemented("no dialer")
 }
 
-// Config returns TLS configuration used to connect to Auth.
-func (c *IdentityCreds) Config() (*tls.Config, error) {
+// TLSConfig returns TLS configuration used to connect to Auth.
+func (c *IdentityCreds) TLSConfig() (*tls.Config, error) {
 	identityFile, err := ReadIdentityFile(c.path)
 	if err != nil {
 		return nil, trace.BadParameter("identity file could not be decoded: %v", err)
 	}
 
-	tlsConfig, err := identityFile.TLS()
+	tlsConfig, err := identityFile.TLSConfig()
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 
 	return configure(tlsConfig), nil
 }
 
+// SSHClientConfig returns SSH configuration used to connect to Proxy.
+func (c *IdentityCreds) SSHClientConfig() (*ssh.ClientConfig, error) {
+	identityFile, err := ReadIdentityFile(c.path)
```

Comment:
Right now this is only being called by client initialization, so it's only called once. In the future I will be adding auto-reload functionality, at which point some type of state will be necessary.

Reply:
Is there a disadvantage to doing
```go
type IdentityCreds struct {
  identityFile *IdentityFile
}
```
(or similar) right away?

Reply:
If it was called immediately in `LoadIdentityFile`, the error would need to be stored and checked by `client.New`. This approach was already turned down in a previous PR.

However, I think you are right that the IdentityFile should be stored after first load to prevent reading again by `SSHConfig()`. There isn't currently a benefit to doing the same for other Credentials, but I'll continue to keep state storage in mind